### PR TITLE
Reduce right-side DOM size

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
This is another follow-up of https://github.com/rust-lang/rust/pull/100429 but not in code blocks this time.

So the idea is: if there is only one element in the `.rightside` element, there is no need to wrap it, we can just create one node.

On each page, I run this JS: `document.getElementsByTagName('*').length`. Important to note: the bigger the number of elements inside the page, the greater the gain. It also doesn't work very nicely on std docs because there are a lot of version annotations. So with this PR, It allows to get the following results:

| file name | before this PR | with this PR | diff |
|-|-|-|-|
| std/default/trait.Default.html | 2189 | 1331 | 39.2% |
| std/vec/struct.Vec.html | 14073 | 13842 | 1.7% |
| std/fmt/trait.Debug.html | 5313 | 4907 | 7.7% |
| std/ops/trait.Index.html | 642 | 630 | 1.9% |
| gtk4/WidgetExt | 3269 | 3061 | 6.4% |

You can test it [here](https://rustdoc.crud.net/imperio/reduce-rightsize-dom-size/gtk4/prelude/trait.WidgetExt.html).

r? @notriddle

### CODE REVIEW

**Review by notriddle — CHANGES_REQUESTED** [review]

File: src/test/rustdoc-gui/anchors.goml [hunk 1]
```diff
@@ -13,10 +13,31 @@ reload:
 assert-css: ("#toggle-all-docs", {"color": "rgb(0, 0, 0)"})
 assert-css: (".fqn .in-band a:nth-of-type(1)", {"color": "rgb(0, 0, 0)"})
 assert-css: (".fqn .in-band a:nth-of-type(2)", {"color": "rgb(173, 55, 138)"})
-assert-css: (".srclink", {"color": "rgb(56, 115, 173)"})
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/test/rustdoc-gui/anchors.goml:16]: Sure!

File: src/librustdoc/html/render/mod.rs [hunk 2]
```diff
@@ -1681,23 +1694,29 @@ fn render_rightside(
...
         const_stability,
         containing_item.stable_since(tcx),
         const_stable_since,
+        if has_src_ref { "" } else { " rightside" },
     );
-    let mut srclink = Buffer::empty_from(w);
-    write_srclink(cx, item, &mut srclink);
-    if has_stability && !srclink.is_empty() {
-        rightside.write_str(" · ");
+    if let Some(l) = cx.src_href(item) {
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/render/mod.rs:1709]: I'm ashamed of this one...

File: src/librustdoc/html/render/mod.rs [hunk 3]
```diff
@@ -840,12 +834,13 @@ fn assoc_method(
...
 /// will include the const-stable version, but no stable version will be emitted, as a natural
 /// consequence of the above rules.
-fn render_stability_since_raw(
+fn render_stability_since_raw_with_extra(
     w: &mut Buffer,
     ver: Option<Symbol>,
     const_stability: Option<ConstStability>,
     containing_ver: Option<Symbol>,
     containing_const_ver: Option<Symbol>,
+    extra: &str,
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/render/mod.rs:843]: :+1:

**Comment by GuillaumeGomez** [comment]
Fix the bad double call and extended GUI tests.

**Comment by Kobzol** [comment]
Doc size wins from the rollup where this PR was merged (probably coming from this PR): https://perf.rust-lang.org/compare.html?start=8a13871b69924b74cfa1d737f2894068b37ea7ea&end=42fa8ac7236f4f78a82aeea543bdd445a59f02e0&stat=size%3Adoc_bytes

**Pull request merged**